.github/workflows/test.yml

```diff
@@ -79,3 +83,65 @@ jobs:
       env:
         LOCKFILE: gun1
       run: mix test test/tesla/adapter/gun_test.exs --trace
+
+  dialyzer:
```

Comment:
Run this only for the `main` matrix (maybe we should rename it from `lint` to `main_version` or something like that.

Reply:
So you just want to run it on a single version, such as Elixir 1.16 for now?

Are you also suggesting that this job is combined with the main Test job? I wouldn't recommend that, but I'll make the change if that's what you want.

Reply:
> So you just want to run it on a single version,
Yes, no matrix setup

> I wouldn't recommend that

I do not have strong opinions, but I would like to hear your thoughts, so I can learn. Eitherway is fine by me

Reply:
Mainly that Dialyzer can be slow -- it takes 2 minutes to run, which is longer than the full test run. It's not so bad here on Tesla though, since it's a small library and not an entire application, but I think it's still nice to run them in parallel.

The other consideration is that this specific fix is for Dialyzer on Elixir 1.16, which we can see is passing. However, the standard test suite isn't yet passing on Elixir 1.16 due to httparrot.

So that may be a reason to keep them separate for now, even if we rather run them sequentially later on.

Reply:
matrix removed for dialyzer.